api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -389,3 +397,455 @@ func (m *AzureManagedMachinePool) validateKubeletConfig() error {
 	}
 	return nil
 }
+
+// validateLinuxOSConfig enforces AKS API configuration for Linux OS custom configuration
+// See: https://learn.microsoft.com/en-us/azure/aks/custom-node-configuration#linux-os-custom-configuration for detailed information.
+func (m *AzureManagedMachinePool) validateLinuxOSConfig() error {
+	if m.Spec.LinuxOSConfig == nil {
+		return nil
+	}
+
+	if m.Spec.LinuxOSConfig.SwapFileSizeMB != nil {
+		minSwapFileSizeMB := int32(1)
+		if *m.Spec.LinuxOSConfig.SwapFileSizeMB < minSwapFileSizeMB {
+			return field.Invalid(
```

Comment:
I think it could be helpful to use the `allErrs` pattern here in case more than one field is invalid.

Reply:
[Example](https://github.com/willie-yao/cluster-api-provider-azure/blob/364f124427e3444481942bf53a98eb6d840c00e5/api/v1beta1/azuremachine_webhook.go#L178)

Reply:
I was following the pattern being followed by the [validators in ValidateCreate](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/b27c4caed7d33352476047f17876fa7f6463f833/api/v1beta1/azuremanagedmachinepool_webhook.go#L73-L81) ; Pattern being: error out immediately.

But I like the `allErrs` pattern too; validate all, collect and report all. 
However, it might affect webhooks performance? 

@jackfrancis @CecileRobertMichon Is there a error catching pattern on ValidateCreate vs ValidateUpdate webhooks we follow?

Reply:
^ @nojnhuh What do you say?